src/NServiceBus.TransportTests/When_stopping.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_stopping : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_allow_message_processing_to_complete(TransportTransactionMode transactionMode)
+        {
+            var messageProcessingStarted = new TaskCompletionSource<bool>();
+            var messageProcessingCancelled = new TaskCompletionSource<bool>();
+
+            OnTestTimeout(() =>
+            {
+                messageProcessingStarted.SetCanceled();
+                messageProcessingCancelled.SetCanceled();
+            });
+
+            await StartPump(
+                async (_, cancellationToken) =>
+                {
+                    messageProcessingStarted.SetResult(true);
+
+                    try
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(10), cancellationToken);
```

Comment:
This will slow down the test suite with 10s right? Is there a way to avoid that?

Reply:
Correct. I spent a good 1-2 hours trying to think of a way to avoid it, but failed.